src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int DEFAULT_INCREMENT_VALUE_OF_ARRAY = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int)(table.length * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value);
+        if (size + 1 == threshold) {
+            resize();
+        }
+        int index = getIndex(newNode);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K,V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, newNode.key)) {
+                currentNode.value = newNode.value;
+                return;
+            } else if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            Node<K, V> currentNode = node;
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(Node<K, V> node) {
+        return node.key == null ? 0 : Math.abs(node.key.hashCode()) % table.length;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * DEFAULT_INCREMENT_VALUE_OF_ARRAY];
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            Node<K, V> currentNode = node;
+            while (currentNode != null) {
+                put(currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+
```

Comment:
Let's create individual method for this piece of code.
```suggestion
        transferOfNodes(oldTable);

```